pkg/image/api/types.go

```diff
@@ -60,6 +77,41 @@ type ImageLayer struct {
 	Size int64
 }
 
+// SignatureValidationResult holds a result of signature validation.
+type SignatureValidationResult string
+
+// ImageSignature holds a signature of an image.
+type ImageSignature struct {
+	// Describes a type of stored blob. Provided by caller.
+	Type string
+	// An opaque binary string which is an image's signature.
+	Content []byte
+	// A time of last server-side validation.
+	ValidatedOn unversioned.Time
+	// Result of validation performed at ValidatedOn.
+	ValidationResult SignatureValidationResult
+
+	// At most one of the following pointers can point to details. If the signature type isn't recognized,
+	// they will all be nil.
+
+	// Details for signature of type "atomic".
+	AtomicSignatureDetails *AtomicSignatureDetails
+	// For any future supported signature type, add additional pointer to its own details struct.
+}
+
+// AtomicSignatureDetails holds metadata of recognized signature of type "atomic". They are parsed from
+// signature's content.
+type AtomicSignatureDetails struct {
+	// Digest of signed manifest.
+	SignedImageName string
+	// Docker image reference
+	SignedDockerImageReference string
+	// Optional, if specified, it is an identity of signer.
+	Creator string
```

Comment:
See the gist with a signature example: https://gist.github.com/mtrmac/a39d184e25dde95012719cc0691ca074#file-signature-template-js-L12

It doesn't necessarily refer to person, it may be a name/version of signing service. The value here is meant for display only via console or in the UI. If we want it structured, we'll pose additional restrictions on the signature format.